esmvalcore/preprocessor/_time.py

```diff
@@ -378,6 +381,40 @@ def extract_season(cube: Cube, season: str) -> Cube:
         coords_to_remove.append("season_year")
 
     result = cube.extract(iris.Constraint(clim_season=season))
+
+    if full:  # remove incomplete seasons
+        iris.coord_categorisation.add_month_number(
+            result, "time", name="month_number"
+        )
+
+        # end of season
+        send = sstart + len(season) - 1
+        if send >= 12:
+            send -= 12
+
+        # add coordinate which flags whether months are part of a full season
+        full_season = np.full_like(result.coord("month_number").points, 1)
+        first_month = np.where(
+            result.coord("month_number").points == sstart + 1
+        )[0][0]
+        last_month = np.where(result.coord("month_number").points == send + 1)[
+            0
+        ][-1]
+        full_season[:first_month] = 0
+        full_season[last_month + 1 :] = 0
+
+        full_season_coord = iris.coords.AuxCoord(
+            full_season, long_name="full_season"
+        )
+        result.add_aux_coord(
+            full_season_coord, data_dims=result.coord_dims("time")
+        )
+
+        result = result.extract(iris.Constraint(full_season=1))
```

Comment:
Personal preference, rather than need, but I would add a logger message in the debug format which months have been removed. Also on a similar note, in case the values are existing in a timeseries, but have been masked out, as far as I understand they won't be removed. I would recommend to add this information somewhere.